internal/testing/migrations_test.go

```diff
@@ -0,0 +1,116 @@
+package testing
+
+import (
+	"bytes"
+	"context"
+	"errors"
+	"testing"
+	"time"
+
+	"go.sia.tech/renterd/internal/consensus"
+	rhpv2 "go.sia.tech/renterd/rhp/v2"
+	"go.uber.org/zap"
+	"go.uber.org/zap/zapcore"
+	"lukechampine.com/frand"
+)
+
+func TestMigrations(t *testing.T) {
+	if testing.Short() {
+		t.SkipNow()
+	}
+
+	// create a new test cluster
+	cluster, err := newTestCluster(t.TempDir(), zap.New(zapcore.NewNopCore()))
+	if err != nil {
+		t.Fatal(err)
+	}
+	defer func() {
+		if err := cluster.Shutdown(context.Background()); err != nil {
+			t.Fatal(err)
+		}
+	}()
+
+	// convenience variables
+	cfg := defaultAutopilotConfig.Contracts
+	w := cluster.Worker
+	b := cluster.Bus
+
+	// add hosts
+	hosts, err := cluster.AddHosts(int(cfg.Hosts))
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	// wait until we have contracts
+	if _, err := cluster.WaitForContracts(); err != nil {
+		t.Fatal(err)
+	}
+
+	// add an object
+	data := make([]byte, rhpv2.SectorSize*4)
+	frand.Read(data)
+	if err := w.UploadObject(bytes.NewReader(data), "foo"); err != nil {
+		t.Fatal(err)
+	}
+
+	t.Helper()
```

Comment:
These need to go inside the helper functions I think. Because they mark the calling function a helper.